ui/jobs/combo_tracker.ts

```diff
@@ -41,33 +56,33 @@ export default class ComboTracker {
       this.AbortCombo(id);
   }
 
-  StateTransition(id, nextState) {
+  StateTransition(id?: string, nextState?: StartMap[string]): void {
     window.clearTimeout(this.comboTimer);
     this.comboTimer = undefined;
 
-    this.isFinalSkill = nextState && Object.keys(nextState.next).length === 0;
-    if (nextState === null || this.isFinalSkill) {
+    this.isFinalSkill = (nextState && Object.keys(nextState?.next ?? {}).length === 0) ?? false;
```

Comment:
```suggestion
    this.isFinalSkill = (nextState && Object.keys(nextState.next).length === 0) ?? false;
```

I think (?) you don't need all of the optional chaining here.